src/main/java/org/zalando/baigan/service/RepositoryFactory.java

```diff
@@ -0,0 +1,44 @@
+package org.zalando.baigan.service;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import org.zalando.baigan.service.aws.S3ConfigurationRepositoryBuilder;
+
+import java.util.List;
+
+@Component
+public class RepositoryFactory {
+
+    private final ConfigurationParser configurationParser;
+
+    @Autowired
+    public RepositoryFactory(final ConfigurationParser configurationParser) {
+        this.configurationParser = configurationParser;
+    }
+
+    public S3ConfigurationRepositoryBuilder s3ConfigurationRepository() {
+        return new S3ConfigurationRepositoryBuilder(configurationParser);
+    }
+
+    public FileSystemConfigurationRepositoryBuilder fileSystemConfigurationRepository() {
+        return new FileSystemConfigurationRepositoryBuilder(configurationParser);
+    }
+
+    public EtcdConfigurationRepository etcdConfigurationRepository(String etcdUrl) {
+        return new EtcdConfigurationRepository(etcdUrl, configurationParser);
```

Comment:
Is there a reason why this one is an actual repository (while the other we have migrated to _builders_)? Worth to also provide a builder - for the sake of symmetry - here or does it not make sense before migrating the Etcd repository to a modern version?

Reply:
Done in https://github.com/zalando-stups/baigan-config/pull/84